ocw/lib/provider.py

```diff
@@ -35,7 +35,9 @@ def needs_to_delete_image(self, order_number, image_date):
         if self.older_than_min_age(image_date):
             max_images_per_flavor = self.cfgGet('cleanup', 'max-images-per-flavor')
             max_image_age = image_date + timedelta(hours=self.cfgGet('cleanup', 'max-image-age-hours'))
-            return order_number > max_images_per_flavor or max_image_age < datetime.now(timezone.utc)
+            # order_number is starting from 0 and max_images_per_flavor is **amount**
+            # so we need to increase order_number by 1 to be compare them
+            return order_number+1 > max_images_per_flavor or max_image_age < datetime.now(timezone.utc)
```

Comment:
I think the most interesting thing is, that the image list must be ordered DESC and starting at index 0 is common.
```suggestion
            return order_number >= max_images_per_flavor or max_image_age < datetime.now(timezone.utc)
```

Reply:
so you think no need to explanation here ?